src/app/components/local-plugin.js

```diff
@@ -0,0 +1,116 @@
+/* global localStorage */
+const yo = require('yo-yo')
+const modalDialog = require('../ui/modaldialog')
+
+module.exports = class LocalPlugin {
+
+  /**
+   * Open a modal to create a local plugin
+   * @param {{profile: any, api: any}[]} plugins The list of the plugins in the store
+   * @returns {Promise<{api: any, profile: any}>} A promise with the new plugin profile
+   */
+  open (plugins) {
+    this.profile = JSON.parse(localStorage.getItem('plugins/local')) || { notifications: {} }
+    return new Promise((resolve, reject) => {
+      modalDialog('Local Plugin', this.form(plugins),
+        { fn: () => resolve(this.create()) },
+        { fn: () => resolve() }
+      )
+    })
+  }
+
+  /**
+   * Create the object to add to the plugin-list
+   */
+  create () {
+    const profile = {
+      ...this.profile,
+      icon: 'data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4KPHN2ZyB3aWR0aD0iMTc5MiIgaGVpZ2h0PSIxNzkyIiB2aWV3Qm94PSIwIDAgMTc5MiAxNzkyIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxwYXRoIGQ9Ik0xMjYyIDEwNzVxLTM3IDEyMS0xMzggMTk1dC0yMjggNzQtMjI4LTc0LTEzOC0xOTVxLTgtMjUgNC00OC41dDM4LTMxLjVxMjUtOCA0OC41IDR0MzEuNSAzOHEyNSA4MCA5Mi41IDEyOS41dDE1MS41IDQ5LjUgMTUxLjUtNDkuNSA5Mi41LTEyOS41cTgtMjYgMzItMzh0NDktNCAzNyAzMS41IDQgNDguNXptLTQ5NC00MzVxMCA1My0zNy41IDkwLjV0LTkwLjUgMzcuNS05MC41LTM3LjUtMzcuNS05MC41IDM3LjUtOTAuNSA5MC41LTM3LjUgOTAuNSAzNy41IDM3LjUgOTAuNXptNTEyIDBxMCA1My0zNy41IDkwLjV0LTkwLjUgMzcuNS05MC41LTM3LjUtMzcuNS05MC41IDM3LjUtOTAuNSA5MC41LTM3LjUgOTAuNSAzNy41IDM3LjUgOTAuNXptMjU2IDI1NnEwLTEzMC01MS0yNDguNXQtMTM2LjUtMjA0LTIwNC0xMzYuNS0yNDguNS01MS0yNDguNSA1MS0yMDQgMTM2LjUtMTM2LjUgMjA0LTUxIDI0OC41IDUxIDI0OC41IDEzNi41IDIwNCAyMDQgMTM2LjUgMjQ4LjUgNTEgMjQ4LjUtNTEgMjA0LTEzNi41IDEzNi41LTIwNCA1MS0yNDguNXptMTI4IDBxMCAyMDktMTAzIDM4NS41dC0yNzkuNSAyNzkuNS0zODUuNSAxMDMtMzg1LjUtMTAzLTI3OS41LTI3OS41LTEwMy0zODUuNSAxMDMtMzg1LjUgMjc5LjUtMjc5LjUgMzg1LjUtMTAzIDM4NS41IDEwMyAyNzkuNSAyNzkuNSAxMDMgMzg1LjV6Ii8+PC9zdmc+',
+      methods: [],
+      events: []
+    }
+    if (!this.profile.name) throw new Error('Plugin should have a name')
```

Comment:
these throws should be catched at some point, right? cause now it seems to silently fails (but i suppose that's fine to merge that one now for an alpha  if we are aware of the left TODOs)

Reply:
When you throw an unhandle error it goes into a console.error. But it's better if it handled indeed.
I just add handlers into the plugin-manager-component, but just a console.log. Maybe it would be better to have an alert instead of a console.log. Which method should I use for that ?

Reply:
just TODO for now is fine. It could either alert or an error in the modal